### DISCUSSION TITLE
Reading transmission data/wrapper?

### DISCUSSION THREAD

**Discussion by seanleblancicdtech** [prowide/prowide-iso20022#84] [General]
I'm curious about the approach others take when wanting to read the wrapper/transmission metadata information that a given MX message might be wrapped in?

Does prowide support a way to do this directly, or are other methods of reading that encouraged?

**Comment by ptorres-prowide** [comment]
Are you referring to the transmission metadata (e.g., routing, sender/receiver information, and security details) outside the core XML content of the MX message, or are you asking about metadata embedded within the XML structure itself, or any other type of metadata related to the MX message?

**Comment by ptorres-prowide** [comment]
Prowide ISO 20022 does support reading message wrapper/metadata, depending on what you mean.

* **Business Application Header (BAH)** — the header that travels with the message:
  `mx.getAppHdr()` returns the parsed AppHdr, for example `BusinessAppHdrV02`, giving you access to `Fr`, `To`, `BizMsgIdr`, `MsgDefIdr`, `CreDt`, etc. You can also build or parse headers using `AppHdrFactory`.

* **Derived metadata for storage/routing** — such as sender, receiver, reference, and message type:
  use `MxSwiftMessage`, whose extraction is customizable through an `MxMetadataStrategy`. By default, it uses `DefaultMxMetadataStrategy`.
  `MxId` identifies the message type.

* **SWIFT network/transmission envelopes external to the XML** — transport-level framing:
  these are out of scope of the MX payload model and are handled at the transport layer.

If you can share a sample of the wrapper you are dealing with, we can point you to the exact API.

Closing for now as inactive — feel free to reopen with a sample.
